test/helpers/convertDataToBytes.js

```diff
@@ -0,0 +1,12 @@
+const convertDataToBytes = (data) => {
+  let concatenatedString = ''
+
+  Object.keys(data).forEach((key) => {
+    concatenatedString += `${data[key]}:::`
+  })
+
+  const buffer = Buffer.from(concatenatedString.substring(0, concatenatedString.length - 3))
```

Comment:
Instead of looping over the object keys, building up a string, and doing an awkward `.substring()` like this you could just do:

```javascript
const concatenatedString = Object.values(data).join(':::')
```

Reply:
Now that I think about it, it probably would be easier just to straight up `JSON.stringify()` the data instead of having an "implied order" (i.e. relying on the first bit before the delimiter to be the providerId.) At the end of the day, the serialization of data is provider-specific anyway so I suppose it doesn't matter. What you do in these tests doesn't necessary have to match what we do in the API.

Reply:
Let me do some measurements on gas costs. If it's insignificant then I'm all for just doing JSON.stringify. Way better than having our own custom schema.